### ISSUE TITLE
fides push should push custom taxonomy first

### ISSUE THREAD

**Issue by mfbrown** [ethyca/fides#2959]
When you do a fides push, it doesn't seem like there is a preferred order in which resources are pushed. This can lead to failures if a system or dataset is pushed before custom taxonomy extensions, and relies on a custom taxonomy element. 

I think it would be better if fides push looked for any taxonomy extensions and always pushed those first, then pushed systems, datasets, etc.

**Comment by rsilvery** [comment]
@mfbrown  is this still an issue?

**Comment by mfbrown** [comment]
Yes. Anytime using Fides push to push a large manifest file of multiple types of Fides resources, there's no guarantee that taxonomy is pushed first.

**Comment by rsilvery** [comment]
What's the impact?

**Comment by mfbrown** [comment]
Whenever systems or datasets that reference custom taxonomy elements are pushed first, the whole push fails, since you can't reference taxonomy elements that don't exist.
